### PR TITLE
RST-1473: Upload Testing

### PR DESCRIPTION
**Pull request by Stivaros**
Uses the new minio server to test whether files are uploaded successfully

### CODE REVIEW

File: spec/features/fill_in_additional_information_page_spec.rb [hunk 1]
```diff
@@ -19,15 +19,21 @@
...
     additional_information_page.load
     additional_information_page.next
 
     expect(confirmation_of_supplied_details_page.confirmation_of_additional_information_answers).not_to have_link(t('components.confirmation_of_supplied_details.remove_file_link'), href: remove_rtf_path)
   end
+
+  scenario "correctly will leave a file in the s3 bucket" do
+    additional_information_page.load
+
+    given_i_am(:company01)
```
Comment (garytaylor) [inline spec/features/fill_in_additional_information_page_spec.rb]: Should this not be first ?  Its the 'setup' phase of the test isnt it ?

File: spec/features/fill_in_additional_information_page_spec.rb [hunk 2]
```diff
@@ -19,15 +19,21 @@
...
+
+    answer_upload_additional_information_question
+    additional_information_page.next
+
+    expect(keys_in_bucket).to include Store.last.hash_store[:additional_information_answers][:upload_additional_information]
```
Comment (garytaylor) [inline spec/features/fill_in_additional_information_page_spec.rb]: You are testing against internals of the application here.  Should you not be testing that the API was sent the correct key and that it is in the bucket ?  Noone cares about Store.last.hash_store really ?

File: spec/features/fill_in_additional_information_page_spec.rb (hunk 1 above)
Comment (Stivaros) [reply spec/features/fill_in_additional_information_page_spec.rb]: As discussed - this is the way I've done it in the other specs. Happy to change if it ever becomes an issue

File: spec/features/fill_in_additional_information_page_spec.rb (hunk 2 above)
Comment (Stivaros) [reply spec/features/fill_in_additional_information_page_spec.rb]: That test is in the stubbed API submission in the `fill_in_whole_form` spec. This is a weird hybrid unit test thing. Perhaps @sivvoy will have a better solution?

**Pull request merged**